### PR TITLE
Feature 3.6/arangobench by duration

### PR DESCRIPTION
**Pull request by dothebart**
backport of https://github.com/arangodb/arangodb/pull/12997

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12682/

**Comment by dothebart** [comment]
tests blue.

File: arangosh/Benchmark/BenchFeature.cpp [hunk 1]
```diff
@@ -287,6 +335,9 @@ void BenchFeature::start() {
         requestTime,
     });
     for (size_t i = 0; i < static_cast<size_t>(_concurrency); ++i) {
+      if (_duration != 0) {
+        _realOperations = threads[i]->_counter;
```
Comment (jsteemann) [inline arangosh/Benchmark/BenchFeature.cpp:339]: Assigning to the same value in a loop...
Does this need to be a `+=`, by the way?
The intent of the code is not clear to me, at least.

Comment (dothebart) [reply arangosh/Benchmark/BenchFeature.cpp:339]: yes, definitely!

Comment (dothebart) [reply arangosh/Benchmark/BenchFeature.cpp]: fixed.

**Review by jsteemann — APPROVED** [review]

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12702/

**Review by KVS85 — CHANGES_REQUESTED** [review]
CHANGELOG entry is required (new options appeared).

**Comment by KVS85** [comment]
```
/Users/jenkins/vm07-mac/oskar/work/ArangoDB/lib/Maskings/AttributeMasking.cpp:51:20: error: loop variable 'entry' is always a copy because the range of type '(anonymous namespace)::VPackObjectIterator' (aka 'arangodb::velocypack::ObjectIterator') does not return a reference [-Werror,-Wrange-loop-analysis]
  for (auto const& entry : VPackObjectIterator(def, false)) {
                   ^
/Users/jenkins/vm07-mac/oskar/work/ArangoDB/lib/Maskings/AttributeMasking.cpp:51:8: note: use non-reference type 'arangodb::velocypack::ObjectIteratorPair'
  for (auto const& entry : VPackObjectIterator(def, false)) {
       ^~~~~~~~~~~~~~~~~~~
1 error generated.
```
At macOS 10.15 (`(clang-1200.0.32.21`).

**Comment by KVS85** [comment]
Tests blue except unrelated clang-12 compilation issue.

**Review by KVS85 — APPROVED** [review]
LGTM.

**Pull request merged**